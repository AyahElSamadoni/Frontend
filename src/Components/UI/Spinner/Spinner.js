import React from "react";

import "./Spinner.css";

const Spinner = () => (
  <div className="atom-spinner">
    <div className="spinner-inner">
      <div className="spinner-line" />
      <div className="spinner-line" />
      <div className="spinner-line" />

      <div className="spinner-circle">&#9679;</div>
    </div>
  </div>
);

export default Spinner;
