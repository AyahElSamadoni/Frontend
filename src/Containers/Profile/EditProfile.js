import React, { Component } from "react";
import { connect } from "react-redux";
import propTypes from "prop-types";
import InputProfile from "../../../Components/UI/InputProfile/InputProfile";
import { editUser } from "../../../Actions/editProfileActions";
import { getProfile } from "../../../Actions/editProfileActions";
import { editImage } from "../../../Actions/editProfileActions";
import Button from "../../../Components/UI/button//button";
import { withRouter } from "react-router-dom";
import { loginUser } from "../../../Actions/authActions";

import "./EditProfile.css";
import Radium, { StyleRoot } from "radium";
/**
 * This is a description of the editProfile Component.
 * @class
 * */
class editProfile extends Component {
  /**
   * @property {json}  editProfileForm               - The Signup Form object that contains all the form's components properties.
   * @property {json} editProfileForm.screen_name  -User's screen name properties and configurations.
   * @property {string} editProfileForm.screen_name.elementType   -The type of the input element.
   * @property {json} editProfileForm.screen_name.elementConfig   -The input element configurations.
   * @property {string} editProfileForm.screen_name.elementConfig.type     -The input element type.
   * @property {string} editProfileForm.screen_name.elementConfig.placeholder  -The input element's placeholder.
   * @property {string} editProfileForm.screen_name.value  -The value of the Screen Name.
   * @property {boolean} editProfileForm.screen_name.valid  -indicates whether the input value is valid or not.
   * @property {boolean} editProfileForm.screen_name.touched  -indicates whether the the input element has been touched or not.
   * @property {boolean} editProfileForm.screen_name.autoFocus  - Is set to true when the element should be autofocused.
   * @property {string} editProfileForm.screen_name.errorMessage  -The error message to be displayed when the input is invalid.
   * @property {json} editProfileForm.screen_name.validation  -The validations required for the screen name.
   * @property {boolean} editProfileForm.screen_name.validation.required  -Indicates whether the field is necessary or not to signup.
   * @property {boolean} editProfileForm.screen_name.validation.nospace  -Is set to true when spaces are not allowed in the input value.
   * @property {boolean} editProfileForm.screen_name.validation.startLetter  -Is set to true when the input value should start with a letter only.
   * @property {integer} editProfileForm.screen_name.validation.maxLength  -The maximum length for the input value.
   * @property {json} editProfileForm.name  -User's name properties and configurations.
   * @property {string} editProfileForm.name.elementType  -The type of the element.
   * @property {json} editProfileForm.name.elementConfig  -The input element configurations.
   * @property {string} editProfileForm.name.elementConfig.type  -The input element type.
   * @property {string} editProfileForm.name.elementConfig.placeholder  -The input element's placeholder.
   * @property {string} editProfileForm.name.value  -The value of the Username.
   * @property {boolean} editProfileForm.name.valid  -indicates whether the input value is valid or not.
   * @property {boolean} editProfileForm.name.touched  -indicates whether the the input element has been touched or not.
   * @property {string} editProfileForm.name.errorMessage -The error message to be displayed when the input is invalid.
   * @property {json} editProfileForm.name.validation  -The validations required for the name.
   * @property {boolean} editProfileForm.name.validation.required  -Indicates whether the field is necessary or not to signup.
   * @property {integer} editProfileForm.name.validation.maxLength  -The maximum length for the input value.
   * @property {json} editProfileForm.bio  -User's bio properties and configurations.
   * @property {string} editProfileForm.bio.elementType  -The type of the element.
   * @property {json} editProfileForm.bio.elementConfig  -The input element configurations.
   * @property {string} editProfileForm.bio.elementConfig.type  -The input element type.
   * @property {string} editProfileForm.bio.elementConfig.placeholder -The input element's placeholder.
   * @property {string} editProfileForm.bio.value -The value of the Username.
   * @property {boolean} editProfileForm.bio.valid -indicates whether the input value is valid or not.
   * @property {boolean} editProfileForm.bio.touched -indicates whether the the input element has been touched or not.
   * @property {string} editProfileForm.bio.errorMessage -The error message to be displayed when the input is invalid.
   * @property {json} editProfileForm.bio.validation -The validations required for the bio.
   * @property {boolean} editProfileForm.bio.validation.required -Indicates whether the field is necessary or not to signup.
   * @property {integer} editProfileForm.bio.validation.maxLength -The maximum length for the input value.
   * @property {string} loc -location of user.
   * @property {bool} formIsValid -indicates whether the form is valid or not before submitting.
   * @property {bool} loading -indicates whether the spinner should be displayed or not.
   * @property {string} file -image file path.
   * @property {string} imapePreview  -default image url.
   * @property {JSON} currentProfile -data of current user.
   */

  state = {
    editProfileForm: {
      screen_name: {
        elementType: "input",
        elementConfig: {
          type: "text",
          placeholder: "Enter your updated screen name"
        },
        value: "",
        validation: {
          required: true,
          nospace: true,
          maxLength: 15,
          startLetter: true
        },
        valid: false,
        touched: false,
        autoFocus: true,
        errorMessage:
          "The screen name should start with a letter and with no spaces",
        invalidScreenname: false
      },
      name: {
        elementType: "input",
        elementConfig: {
          type: "text",
          placeholder: "Enter your updated name"
        },
        value: "",
        validation: {
          maxLength: 15
        },
        errorMessage: "Maximum length is 15",
        valid: false,
        touched: false
      },

      bio: {
        elementType: "input",
        elementConfig: {
          type: "input",
          placeholder: "Enter your bio "
        },
        value: "",
        validation: {
          maxLength: 100
        },
        errorMessage: "Maximum length is 100",
        valid: false,
        touched: false
      }
    },
    loc: "",
    formIsValid: false,
    loading: false,
    error: {},
    file: "",
    imagePreview: "http://ssl.gstatic.com/accounts/ui/avatar_2x.png",
    currentProfile: {}
    //token: "",
  };
  /**
   * Handles changes in the input elements by running its validity checks and updating its value in the state.
   * @param {event} event - The change event.
   * @param {integer} inputIdentifier - The input element's identifier.
   */
  inputChangeHandler(event, inputIdentifier) {
    const updatedEditsProfileForm = {
      ...this.state.editProfileForm
    };
    const updatedFormElement = {
      ...updatedEditsProfileForm[inputIdentifier]
    };
    updatedFormElement.value = event.target.value;
    updatedFormElement.valid = this.checkValidity(
      updatedFormElement.value,
      updatedFormElement.validation
    );
    updatedFormElement.touched = true;
    updatedEditsProfileForm[inputIdentifier] = updatedFormElement;

    let formIsValid = true;
    for (let inputIdentifier in updatedEditsProfileForm) {
      formIsValid =
        updatedEditsProfileForm[inputIdentifier].valid && formIsValid;
    }
    this.setState({
      editProfileForm: updatedEditsProfileForm,
      formIsValid: formIsValid
    });
  }

  _handleChange = event => {
    this.setState({ loc: event.target.value });
    //this.setState({ location: event.target.value });
  };
  /**
   * Check if a value matches a given set of validation rules.
   * @param {string} value - The value to be checked.
   * @param {json} rules - The set of vaidation rules to check against.
   */
  checkValidity(value, rules) {
    let isValid = true;

    if (rules.startLetter) {
      isValid = !!value.match(/^[a-zA-Z][a-zA-Z_0-9]*$/) && isValid;
    }

    if (rules.nospace) {
      isValid = !(value.indexOf(" ") >= 0) && isValid;
    }
    if (rules.minLength) {
      isValid = value.length >= rules.minLength && isValid;
    }

    if (rules.maxLength) {
      isValid = value.length <= rules.maxLength && isValid;
    }

    return isValid;
  }

  submitHandler = event => {
    event.preventDefault();
    this.setState({ loading: true });
    const formData = {};
    for (let formElementIdentifier in this.state.editProfileForm) {
      formData[formElementIdentifier] = this.state.editProfileForm[
        formElementIdentifier
      ].value;
    }

    const user = {
      screen_name: this.state.editProfileForm.screenname.value,
      name: this.state.editProfileForm.username.value,
      location: this.state.loc,
      bio: this.state.editProfileForm.bio.value
    };
    const img_url = {
      img_url: this.state.imagePreview
    };
    this.props.editImage(img_url);
    this.props.editUser(this.props, user).then(() => {
      this.props.history.push("/profile");
    });
  };

  onChange(e) {
    e.preventDefault();

    let reader = new FileReader();
    let file = e.target.files[0];

    reader.onload = e => {
      this.setState({
        file,
        imagePreview: reader.result
      });
    };
    if (file) {
      reader.readAsDataURL(file);
    }
    console.log(reader.result);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.error) {
      this.setState({ error: nextProps.error }, () => {
        this.setState({ loading: false });
        if (this.state.msg === "screen name already registered.") {
          console.log("hello");
          this.setState({ errorScreenname: true });
        } else if (
          this.state.msg ===
          ' "screen_name" length must be at least 3 characters long'
        ) {
          this.setState({ errorLenScreenname: true });
        } else if (
          this.state.msg ===
          ' "screen_name" length must be less than or equal to 15 characters long'
        ) {
          this.setState({ errorLenScreenname: true });
        }
      });
    }
  }

  render() {
    const formElementArray = [];
    for (let key in this.state.editProfileForm) {
      formElementArray.push({
        id: key,
        config: this.state.editProfileForm[key]
      });
    }
    console.log(this.props.auth.currentUser);
    let form = (
      <form onSubmit={this.submitHandler} className="editBox">
        {formElementArray.map(formElement => (
          <InputProfile
            key={formElement.id}
            elementType={formElement.config.elementType}
            elementConfig={formElement.config.elementConfig}
            value={this.props.auth.currentUser[formElement.id]} //formElement.config.value}
            invalid={!formElement.config.valid}
            errorMessage={formElement.config.errorMessage}
            shouldValidate={formElement.config.validation}
            touched={formElement.config.touched}
            autoFocus={formElement.config.autoFocus}
            changed={event => this.inputChangeHandler(event, formElement.id)}
          />
        ))}

        <div className="col-sm-20">
          <select onChange={this._handleChange} className="form-control">
            <option selected="">Select location</option>

            <option value="Afghanistan">Afghanistan</option>
            <option value="Algeria">Algeria</option>
            <option value="Angola">Angola</option>
            <option value="Argentina">Argentina</option>
            <option value="Armenia">Armenia</option>
            <option value="Australia">Australia</option>
            <option value="Austria">Austria</option>

            <option value="Bahrain">Bahrain</option>
            <option value="Belgium">Belgium</option>
            <option value="Brazil">Brazil</option>
            <option value="Bulgaria">Bulgaria</option>
            <option value="Cameroon">Cameroon</option>
            <option value="China">China</option>
            <option value="Colombia">Colombia</option>
            <option value="Costa Rica">Costa Rica</option>
            <option value="Cuba">Cuba</option>
            <option value="Denmark">Denmark</option>
            <option value="Canada">Canada</option>
            <option value="Ecuador">Ecuador</option>
            <option value="Egypt">Egypt</option>
            <option value="Ethiopia">Ethiopia</option>

            <option value=" Finland"> Finland</option>
            <option value="France">France</option>
            <option value="Germany">Germany</option>
            <option value="Greece">Greece</option>
            <option value="France">France</option>
            <option value="France">France</option>
            <option value="France">France</option>
            <option>United States</option>
            <option>England</option>
            <option>Italy</option>
            <option>Lebanon</option>
          </select>
        </div>
        <div className="form-group">
          <Button>Save</Button>
        </div>
      </form>
    );

    return (
      <div>
        <div>
          <div>
            <h1 className="text-center editHeader">Edit your profile</h1>

            <div className="row">
              <div className="col-sm-5" />
              <div className="col-sm-3">
                <img
                  src={this.state.imagePreview}
                  width="200"
                  height="200"
                  border="10"
                  className="img-rounded"
                />
                <h1> </h1>
                <div class="input-group">
                  <div class="input-group-prepend">
                    <span class="input-group-text" id="inputGroupFileAddon01">
                      Upload
                    </span>
                  </div>
                  <div class="custom-file">
                    <input
                      type="file"
                      class="custom-file-input"
                      id="inputGroupFile01"
                      aria-describedby="inputGroupFileAddon01"
                      onChange={e => this.onChange(e)}
                    />
                    <label class="custom-file-label" for="inputGroupFile01">
                      Choose file
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="text-center">
            <h4> </h4>
            <h4> </h4>
            <h2 className="editHeader">update your info</h2>
            <div className="col-xs-20">
              <div className="container">{form}</div>
            </div>
          </div>

          <div className="row">
            <div className="col-sm-5" />
            <span> </span>
          </div>
        </div>
      </div>
    );
  }
}
const mapStateToProps = state => ({
  auth: state.auth,
  edit: state.edit,
  error: state.errors
});

export default withRouter(
  connect(
    mapStateToProps,
    { loginUser } //{ editUser, editImage, getProfile }
  )(editProfile)
);
