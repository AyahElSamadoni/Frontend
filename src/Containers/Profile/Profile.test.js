import React from "react";
import { shallow, configure, mount } from "enzyme";
import Profile from "./Profile";
import Nav from "./Profile";
import { shallowToJson } from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import Spinner from "../../Components/UI/Spinner/Spinner";
import FanModal from "../../Components/FansBox/Modal/fansModal";
import NovaModal from "../../Components/novaModal/novaModal";
import {
  CSSTransition,
  Transition,
  TransitionGroup
} from "react-transition-group";

configure({ adapter: new Adapter() });
describe("<Profile>", () => {
  it(" profile should be defined", () => {
    expect(<Profile />).toBeDefined();
  });
  it("profile should render", () => {
    const wrapper = shallow(<Profile />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it(" profile should render", () => {
    const wrapper = shallow(<Profile />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render", () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render", () => {
    const wrapper = shallow(<Spinner />);
    expect(wrapper).toMatchSnapshot();
  });
  it("should render", () => {
    const wrapper = shallow(<TransitionGroup />);
    expect(wrapper).toMatchSnapshot();
  });

  test("render a small label", () => {
    const wrapper = shallow(<a className="referencecolor">Follow</a>);
    expect(wrapper).toMatchSnapshot();
  });

  // test("sets loading state to true on save press", () => {
  //     // const toggleFans = jest.fn();
  //     const wrapper = shallow(
  //    <Profile/>
  //     );
  
  //     wrapper.find('[id="1"]').simulate("click");
  //     expect(wrapper.state("modalShownFans")).toEqual(true);
  //   });



  // test("render a document title", () => {
  //   const wrapper = shallow(<FanModal boxName="Followings" />);
  //   expect(wrapper.prop("boxName")).toEqual("Followings");
  // });

  // it("should render", () => {
  //   const wrapper = shallow(<Profile />);
  //   expect(
  //     wrapper.find("button").hasClass("btn profilebtn profile-edit-btn")
  //   ).to.equal(true);
  // });
  // it("should call the onClick function when 'Subtract' button is clicked when the operator is '-'", () => {
  //   const wrapper = shallow(<Profile />);
  //   wrapper.setProps("1");
  //   const deleteClicked = jest.fn();
  //   wrapper.instance().deleteNovaHandler = deleteClicked;
  //   wrapper.find("deleteClicked").onClick();
  //   expect(deleteClicked).toHaveBeenCalledTimes(1);
  // });
  // 
  // it("should render 1 <menu>s", () => {
  //   const wrapper = shallow(<menu>
  //     <TransitionGroup className="d-flex flex-column bd-highlight mb-3 justify-content-center align-items-center">
  //       {novas}
  //     </TransitionGroup>
  //   </menu> );
  //   expect(wrapper.find("menu")).toHaveLength(1);
  // });

  // test("toggle followers", () => {
  //   const value = 12;
  //   const reNovaHandler = jest.fn();
  //   const wrapper = shallow(
  //     <button
  //     className="btn btn-success profilebtn profile-edit-btn"
  //     onClick={this.follow}
  //   />
  //   );

  //   expect(wrapper).toMatchSnapshot();

  //   wrapper.find("span").simulate("change", false);

  //   expect(toggleFollowers).toBeCalledWith(true);
  // });



});
