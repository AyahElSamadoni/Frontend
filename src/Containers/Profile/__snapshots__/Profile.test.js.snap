// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile>  profile should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(profile) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<Profile> profile should render 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<Profile> render a small label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <a
    className="referencecolor"
  >
    Follow
  </a>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Follow",
      "className": "referencecolor",
    },
    "ref": null,
    "rendered": "Follow",
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Follow",
        "className": "referencecolor",
      },
      "ref": null,
      "rendered": "Follow",
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Profile> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(profile) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`<Profile> should render 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Spinner />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="spinner-inner"
      >
        <div
          className="spinner-line"
        />
        <div
          className="spinner-line"
        />
        <div
          className="spinner-line"
        />
        <div
          className="spinner-circle"
        >
          ●
        </div>
      </div>,
      "className": "atom-spinner",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="spinner-line"
          />,
          <div
            className="spinner-line"
          />,
          <div
            className="spinner-line"
          />,
          <div
            className="spinner-circle"
          >
            ●
          </div>,
        ],
        "className": "spinner-inner",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "spinner-line",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "spinner-line",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "spinner-line",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "●",
            "className": "spinner-circle",
          },
          "ref": null,
          "rendered": "●",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="spinner-inner"
        >
          <div
            className="spinner-line"
          />
          <div
            className="spinner-line"
          />
          <div
            className="spinner-line"
          />
          <div
            className="spinner-circle"
          >
            ●
          </div>
        </div>,
        "className": "atom-spinner",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="spinner-line"
            />,
            <div
              className="spinner-line"
            />,
            <div
              className="spinner-line"
            />,
            <div
              className="spinner-circle"
            >
              ●
            </div>,
          ],
          "className": "spinner-inner",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "spinner-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "spinner-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "spinner-line",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "●",
              "className": "spinner-circle",
            },
            "ref": null,
            "rendered": "●",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<Profile> should render 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TransitionGroup
    childFactory={[Function]}
    component="div"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div />,
      "value": Object {
        "isMounting": false,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div />,
        "value": Object {
          "isMounting": false,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
